R/callbacks.R

```diff
@@ -22,36 +22,38 @@ read_progress_log_file <- function(filename, skip = 0) {
 # generic rendering of progress text straight from a set of files
 #' @importFrom utils flush.console
 render_progress <- function(reads) {
-
   reads <- unlist(reads)
   reads[is.na(reads)] <- ""
   some_results <- any(nchar(reads) > 0)
   if (some_results) {
 
     # optionally add blanks to put lines at the edges
-    if (length(reads) > 1)
+    if (length(reads) > 1) {
       reads <- c("", reads, "")
+    }
 
     msg <- paste(reads, collapse = "  |  ")
 
     message("\r", appendLF = FALSE)
     message(msg, appendLF = FALSE)
     flush.console()
-
   }
 }
 
 # callback functions
 percentages <- function() {
-  reads <- lapply(greta_stash$percentage_log_files,
-                  read_progress_log_file)
+  reads <- lapply(
+    greta_stash$percentage_log_files,
+    read_progress_log_file
+  )
   render_progress(reads)
 }
 
 progress_bars <- function() {
   reads <- lapply(greta_stash$progress_bar_log_files,
-                  read_progress_log_file,
-                  skip = 1)
+    read_progress_log_file,
+    skip = 1
```

Comment:
and this is an example of `lapply` not moving things on the first line

Reply:
@lorenzwalthert the style guide for long lines says:
> Short unnamed arguments can also go on the same line as the function name, even if the whole function call spans multiple lines.

But the unnamed arguments here are long, so they would run over the 80 character limit if they were all on the top line. So I would assume they should be on multiple lines in this case.

Reply:
I appreciate this is ambiguous and minor though, so feel free to ignore :)

Reply:
Yes, a 80 character limit is a [long standing feature request](https://github.com/r-lib/styler/issues/247) for styler. The style guide also [says](https://style.tidyverse.org/syntax.html#long-lines)
> You may also place several arguments on the same line if they are closely related to each other, e.g., strings in calls to paste() or stop()

This is also something styler can't do. But since styler is non-invasive, the user's intentional formatting in this case should not be destroyed by styler, so if you hit 80 characters or think the arguments are unrelated, you can break the line and styler should not modify it further.

Reply:
Yep, good point. Thanks for getting back to us, and so promptly!

Reply:
> and this is an example of lapply not moving things on the first line

Ok, now I understand you better 😀. Yes, styler does not remove this line break here to put all unnamed arguments on the first line, because it thinks you deliberately broke it and the style guide does not say *put all unnamed arguments on one line* (-> non-invasive).

Reply:
> Thanks for getting back to us, and so promptly!

I am always exited when great projects like greta start using styler 🎉

Reply:
Thanks so much for helping us understand, Lorenz! I appreciate the design of it as a non-invasive styler, very cool!

Looking forward to merging this PR soon :)